packages/xdl/src/EmbeddedAssets.ts

```diff
@@ -253,35 +263,44 @@ async function _maybeConfigureExpoKitEmbeddedAssetsAsync(config: EmbeddedAssetsC
   }
 }
 
-async function _maybeConfigureExpoUpdatesEmbeddedAssetsAsync(config: EmbeddedAssetsConfiguration) {
+/**
+ * Guess if this is a users first publish and run a slightly modified expo-updates plugin
+ */
+async function _maybeRunModifiedExpoUpdatesPluginAsync(config: EmbeddedAssetsConfiguration) {
   if (!config.pkg.dependencies?.['expo-updates'] || config.target === 'managed') {
     return;
   }
 
-  let isLikelyFirstPublish = false;
-
   const { projectRoot, exp, releaseChannel, iosManifestUrl, androidManifestUrl } = config;
 
   const { iosSupportingDirectory: supportingDirectory } = getIOSPaths(projectRoot);
 
   // iOS expo-updates
-  if (fs.existsSync(path.join(supportingDirectory, 'Expo.plist'))) {
-    // This is an app with expo-updates installed, set properties in Expo.plist
-    await IosPlist.modifyAsync(supportingDirectory, 'Expo', (configPlist: any) => {
-      if (configPlist.EXUpdatesURL === 'YOUR-APP-URL-HERE') {
-        isLikelyFirstPublish = true;
-      }
-      configPlist.EXUpdatesURL = iosManifestUrl;
-      configPlist.EXUpdatesSDKVersion = exp.sdkVersion;
+  let isLikelyFirstIOSPublish = false;
+  const expoPlistPath = path.join(supportingDirectory, 'Expo.plist');
+  if (fs.existsSync(expoPlistPath)) {
+    const expoPlistForProject = plist.parse(await readFileSync(expoPlistPath, 'utf8'));
+    const currentIOSUpdatesURL = expoPlistForProject[IOSConfig.Updates.Config.UPDATE_URL];
+    if (currentIOSUpdatesURL === PLACEHOLDER_URL) {
+      isLikelyFirstIOSPublish = true;
+
+      // The username is only used for defining a default updates URL.
+      // Since we overwrite the URL below the username is superfluous.
+      IOSConfig.Updates.setUpdatesConfig(exp, expoPlistForProject, /*expoUsername*/ null);
```

Comment:
This edits *more* properties than before, but they are all appropriate and their absence was arguably another defect. Namely:

1. `EXUpdatesCheckOnLaunch`
2. `EXUpdatesLaunchWaitMs`
3. `EXUpdatesEnabled`

Further it will set the `EXUpdatesRuntimeVersion` instead of the `EXUpdatesSDKVersion` when appropriate.

https://github.com/expo/expo-cli/blob/master/packages/config-plugins/src/ios/Updates.ts#L68

And analogously for Android